runtime/src/main/java/org/corfudb/runtime/view/LayoutManagementView.java

```diff
@@ -110,6 +110,34 @@ public void bootstrapNewNode(String endpoint)
         log.info("bootstrapNewNode: New node {} bootstrapped.", endpoint);
     }
 
+    /**
+     * Removes a node from the layout
+     * @param currentLayout the layout to remove the node from
+     * @param endpoint the node to remove
+     */
+    public void removeNode(Layout currentLayout, String endpoint) throws QuorumUnreachableException,
+            CloneNotSupportedException,
+            LayoutModificationException, OutrankedException {
+        currentLayout.setRuntime(runtime);
+        sealEpoch(currentLayout);
+
+        LayoutBuilder builder = new LayoutBuilder(currentLayout);
```

Comment:
We can build the layout before sealing.
So the order can be :
- generate new layout
- seal
- consensus
- reconfigure Sequencer

Reply:
This way if there is an error while building the new layout, you would not have sealed and blocked the entire system.

Reply:
Good point.